pkg/reconciler/accessor/istio/virtualservice.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package istio
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"knative.dev/pkg/apis/istio/v1alpha3"
+	sharedclientset "knative.dev/pkg/client/clientset/versioned"
+	istiolisters "knative.dev/pkg/client/listers/istio/v1alpha3"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/logging"
+)
+
+// VirtualServiceAccessor is an interface for accessing VirtualService.
+type VirtualServiceAccessor interface {
+	GetVirtualServiceClient() sharedclientset.Interface
+	GetVirtualServiceLister() istiolisters.VirtualServiceLister
+}
+
+// ReconcileVirtualService reconciles VirtiualService to the desired status.
+func ReconcileVirtualService(ctx context.Context, owner kmeta.Accessor, desired *v1alpha3.VirtualService,
+	recorder record.EventRecorder, vsAccessor VirtualServiceAccessor) (*v1alpha3.VirtualService, error) {
```

Comment:
Can we standardize on getting things off the context here?

https://github.com/knative/serving/blob/1e95561100d301dc0db945ab3c7de70abb42aa4b/pkg/reconciler/reconciler.go#L108

Reply:
I just want to make sure I understand this correctly.

So we should get EventRecorder from Context in this function. And the caller of this function needs to load EventRecorder to Context. Is my understanding correct?